src/NServiceBus.Transport.RabbitMQ/Sending/BasicPropertiesExtensions.cs

```diff
@@ -57,7 +57,11 @@ public static void Fill(this IBasicProperties properties, OutgoingMessage messag
                 }
             }
 
-            if (messageHeaders.TryGetValue(NServiceBus.Headers.ContentType, out var contentType) && contentType != null)
+            if (dispatchProperties.TryGetValue(NativeIntegrationDispatchPropertiesExtensions.ContentTypeAttribute, out var contentType) && contentType != null)
```

Comment:
By looking at the scenarios in ASB and RabbitMQ it seems that [ASB has the possibility to customize the native outgoing message](https://docs.particular.net/transports/azure-service-bus/native-message-access#access-to-the-native-azure-service-bus-outgoing-message) to the needs of the user while the rabbitmq transport doesn't provide that. Has it been considered to provide a general outgoing native message customization opportunity in the transport to unlock similar customization scenarios like ASB has that could also make it possible to override basic property values where desirable?

FYI I'm not a huge fan of the customization in ASB because it is implemented a bit in awkward ways but at least considering some alignment seems warranted